src/capture-eye.ts

```diff
@@ -283,6 +310,39 @@ export class CaptureEye extends LitElement {
     this.setButtonActive(false);
   }
 
+  private handleResize(entry: ResizeObserverEntry) {
+    if (entry.contentRect.width > 0 && entry.contentRect.height > 0) {
+      this.setButtonFullVisibility();
+    }
+  }
+
+  private handleSlotChange(event: Event) {
+    if (!this._resizeObserver) {
+      return;
+    }
+
+    const slot = event.target as HTMLSlotElement;
+    if (!slot) {
+      return;
+    }
+
+    if (!slot.assignedNodes({ flatten: true }).some((node) => {
+      if (node.nodeType !== Node.ELEMENT_NODE) {
+        return false;
+      }
+      const rect = (node as Element).getBoundingClientRect();
+      if (rect.width > 0 && rect.height > 0) {
+        this.setButtonFullVisibility();
+      } else {
+        this._resizeObserver?.observe(node as Element);
+        node.addEventListener('error', () => this.setButtonFullVisibility());
+      }
+      return true;
+    })) {  // No any node
+      this.setButtonFullVisibility();
+    }
```

Comment:
It's a bit difficult to understand the if clause at the first glance so I would suggest flatten the logic with some clause guards. Also the check of `(node)` in `some` should be able to combined with `node.nodeType` check.

```
const nodes = slot.assignedNodes({ flatten: true }).filter(node.nodeType === Node.ELEMENT_NODE);
if (nodes.length === 0) {
    this.setButtonFullVisibility();
    return;
}
nodes.forEach(...)
```

I'd imagine something like this in my mind. What do you think about it?